### PR TITLE
Static linking

### PR DESCRIPTION
**Pull request by pfultz2**
Improve static linking support and add a static build to jenkins.

### CODE REVIEW

**Comment by JehandadKhan** [comment]
Why are we bumping the boost version ?

**Comment by pfultz2** [comment]
> Why are we bumping the boost version ?

So we can use imported targets.

**Comment by daniellowell** [comment]
Is this ready?

**Comment by pfultz2** [comment]
> Is this ready?

Almost, let me fix the hip-clang docker.

**Comment by daniellowell** [comment]
@pfultz2 conflicts and looks like the docker cannot be built.

File: Jenkinsfile [hunk 1]
```diff
@@ -250,6 +250,13 @@ pipeline {
                     }
                 }
 
+                stage('Hip Static Release on /usr/local') {
+                    agent{ label rocmnode("vega") }
+                    steps{
+                        buildJob('hcc', '-DBUILD_DEV=On -DCMAKE_BUILD_TYPE=release -DBUILD_SHARED_LIBS=On', image, "")
```
Comment (daniellowell) [inline Jenkinsfile:256]: Reference.

**Pull request merged**